analyzer/consensus/consensus.go

```diff
@@ -300,6 +284,36 @@ func (m *processor) ProcessBlock(ctx context.Context, uheight uint64) error {
 		return err
 	}
 
+	return nil
+}
+
+// Implements BlockProcessor interface.
+func (m *processor) ProcessBlock(ctx context.Context, uheight uint64) error {
+	if uheight > math.MaxInt64 {
+		return fmt.Errorf("height %d is too large", uheight)
+	}
+	height := int64(uheight)
+
+	// Fetch all data.
+	fetchTimer := m.metrics.BlockFetchLatencies()
+	data, err := fetchAllData(ctx, m.source, m.network, height, m.mode == analyzer.FastSyncMode)
+	if err != nil {
+		if strings.Contains(err.Error(), fmt.Sprintf("%d must be less than or equal to the current blockchain height", height)) {
+			return analyzer.ErrOutOfRange
+		}
+		return err
+	}
+	fetchTimer.ObserveDuration() // We make no observation in case of a data fetch error; those timings are misleading.
```

Comment:
I was just answering a similar thread in #572 :)
IDK that they're misleading. If, for example, the data fetch fails because of a timeout in 90% of the cases, it's very informative to learn that we're spending a minute per block in data fetch, rather than having the chart show the average fetch time for the other 10% of cases.
It's maybe a tiny bit easier to argue that failure cases are misleading when the data fetch fails quickly because of e.g. a malformed request. But even then ... that's objectively the time spent on data fetch, right? The anomaly in the graph will alert us to something.

Reply:
Hmm true, updated!